src/ott/solvers/linear/sinkhorn_lr.py

```diff
@@ -687,7 +682,10 @@ def one_iteration(
         lambda: state.reg_ot_cost(ot_prob, epsilon=self.epsilon),
         lambda: jnp.inf
     )
-    error = state.compute_error(previous_state)
+    error = jax.lax.cond(
```

Comment:
good catch!

Reply:
The convergence criterion (as currently implemented) only makes “theoretical” sense when the gradient step is constant throughout the iterations. In that case, for each choice of gamma, we could adapt the desired approximation error (by considering gamma^2 \times esp) as suggested Marco. This will be equivalent as the current procedure and we are additionally eliminating an indeterminacy.

However, as we are using a rescaled gradient step, we should, in theory, monitor 1/gamma_k^2 (err_Q + err_R + err_g). And if gamma_k^2 converges towards a constant (close to 1) then it becomes equivalent to only monitor the errors. We could also apply the same idea as in the constant gradient-step case, by monitoring gamma_k^2 \times eps instead.

I hope this helps a little.